plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2401,7 +2404,7 @@ private void cleanExtraOutputFiles(ConnectorSession session, String queryId, Str
             FileIterator iterator = fileSystem.listFiles(location);
             while (iterator.hasNext()) {
                 FileEntry file = iterator.next();
-                String fileName = new Path(file.location()).getName();
+                String fileName = getFileName(file.location());
                 if (isFileCreatedByQuery(fileName, queryId) && !filesToKeep.contains(location + "/" + fileName)) {
```

Comment:
appendPath instead of string concatenation

Reply:
This is existing behavior, so we can follow up. There are probably many of these to update, since `appendPath` was added more recently. It's good to note these places that need updating.